apm-agent-plugins/apm-log-shader-plugin/apm-log4j1-plugin/src/main/java/co/elastic/apm/agent/log4j1/Log4j1LogShadingHelper.java

```diff
@@ -0,0 +1,91 @@
+/*-
+ * #%L
+ * Elastic APM Java agent
+ * %%
+ * Copyright (C) 2018 - 2020 Elastic and contributors
+ * %%
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ * #L%
+ */
+package co.elastic.apm.agent.log4j1;
+
+import co.elastic.apm.agent.log.shader.AbstractLogShadingHelper;
+import co.elastic.apm.agent.log.shader.Utils;
+
+import co.elastic.logging.log4j.EcsLayout;
+import org.apache.log4j.FileAppender;
+import org.apache.log4j.RollingFileAppender;
+import org.apache.log4j.WriterAppender;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.annotation.Nullable;
+import java.io.IOException;
+
+class Log4j1LogShadingHelper extends AbstractLogShadingHelper<WriterAppender> {
+
+    private static final Logger logger = LoggerFactory.getLogger(Log4j1LogShadingHelper.class);
+
+    private static final Log4j1LogShadingHelper INSTANCE = new Log4j1LogShadingHelper();
+
+    static Log4j1LogShadingHelper instance() {
+        return INSTANCE;
+    }
+
+    Log4j1LogShadingHelper() {}
+
+    @Override
+    protected String getAppenderName(WriterAppender appender) {
+        return appender.getName();
+    }
+
+    @Override
+    @Nullable
+    protected WriterAppender createAndConfigureAppender(WriterAppender originalAppender, String appenderName) {
+
+        RollingFileAppender shadeAppender = null;
+        if (originalAppender instanceof FileAppender) {
+            try {
+                FileAppender fileAppender = (FileAppender) originalAppender;
+                String shadeFile = Utils.computeShadeLogFilePath(fileAppender.getFile());
+
+                EcsLayout ecsLayout = new EcsLayout();
+                ecsLayout.setServiceName(getServiceName());
```

Comment:
It's more of an idea or a thing to discuss but I think it makes sense to add the `appenderName` to the event dataset so that it's possible to filter by different log streams of an application.
```suggestion
                ecsLayout.setServiceName(getServiceName());
                ecsLayout.setEventDataset(getServiceName() + "." + appenderName);
```

Reply:
Yeah, good idea. I'll apply this suggestion, unless you want to think of an alternative. As long as this is experimental, I think it's OK to start with something that may change later.